src/Workspace.cpp

```diff
@@ -289,17 +311,20 @@ void WorkspaceFolder::initialize()
 void WorkspaceFolder::setupWithConfiguration(const ClientConfiguration& configuration)
 {
     client->sendTrace("workspace: setting up with configuration");
-    isConfigured = true;
-    if (configuration.sourcemap.enabled)
+    platform = LSPPlatform::getPlatform(configuration, &fileResolver, this);
+
+    fileResolver.platform = platform.get();
+
+    if (!isConfigured)
     {
-        client->sendTrace("workspace: sourcemap enabled");
-        if (!isNullWorkspace() && !updateSourceMap())
-        {
-            client->sendWindowMessage(
-                lsp::MessageType::Error, "Failed to load sourcemap.json for workspace '" + name + "'. Instance information will not be available");
-        }
+        isConfigured = true;
```

Comment:
What happens in this case if the `platform.type` was changed midway through an active session. This check looks like we only mutate registered definitions on startup when we are first not configured, so this wouldn't work I think?

Reply:
Actually nvm, I just remembered from the extension code that we need to reload the language server whenever type definitions changes. So this is fine